sqlglot/parser.py

```diff
@@ -4360,7 +4360,25 @@ def _parse_bitwise(self) -> t.Optional[exp.Expression]:
         return this
 
     def _parse_term(self) -> t.Optional[exp.Expression]:
-        return self._parse_tokens(self._parse_factor, self.TERM)
+        this = self._parse_factor()
+
+        while self._match_set(self.TERM):
+            klass = self.TERM[self._prev.token_type]
+            comments = self._prev_comments
+            expression = self._parse_factor()
+
+            this = self.expression(klass, this=this, comments=comments, expression=expression)
+
+            if isinstance(this, exp.Collate):
+                expr = this.expression
+
+                # Preserve collations such as pg_catalog."default" (Postgres) as columns, otherwise
+                # fallback to Identifier / Var
+                if isinstance(expr, exp.Column) and len(expr.parts) == 1:
+                    ident = expr.this
+                    this.set("expression", ident if ident.quoted else exp.var(ident.name))
```

Comment:
Are you sure `Column.this` is always gonna be an identifier here? `ident.quoted` will fail if not.

Reply:
Hmm, can we a parent `Column` without an `Identifier` child? Currently the RHS is parsed either into a (1) `exp.Column(this=exp.Identifier())` or (2) a string `exp.Literal`, depending on the quotes.

Reply:
> Hmm, can we a parent Column without an Identifier child?

I'm not sure, I'd need to analyze the code paths in the parser to tell. Probably not, but if it's easy to be a bit more defensive here it'd be good. Your call.

Reply:
I think we'll only create an `exp.Column` at `_parse_column_reference()`, and that's only if `this` is an `exp.Identifier`.

However, I'll go ahead and add one more branch just in case